test/soldiers.test.js

```diff
@@ -0,0 +1,131 @@
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.statusCode).toBe(201);
+    expect(res.body).toContain('id');
+    expect(res.body).toContain(testSoldier.id);
+    expect(res.body).toContain(testSoldier.name);
+  });
+
+  it('checks that when soldier not created because of duplicate id status is 400', async () => {
+    await addNewSoldier(testSoldier);
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier2,
+    });
+    expect(res.statusCode).toBe(400);
+    expect(res.json()).toEqual({ message: 'soldier not created - this id already exists!' });
+  });
+
+  it('checks for status 500 when theres a server error', async () => {
+    const addNewSoldierSpy = vi.spyOn(database, 'addNewSoldier');
+    addNewSoldierSpy.mockImplementation(() => {
+      const err = new Error();
+      throw err;
+    });
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.json()).toEqual({ message: 'server error' });
+  });
+});
+
+describe('Get soldier by id route', () => {
+  it('checks that when soldier is found status is 200, and that found soldier is sent', async () => {
+    await addNewSoldier(testSoldier);
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers/${testSoldier.id}`,
+    });
+    expect(res.statusCode).toBe(200);
+    expect(res).toBeTypeOf('object');
+    expect(res.body).toContain(testSoldier.name);
+    expect(res.body).toContain(testSoldier.id);
+  });
+
+  it('checks that when soldier is not found status is 404', async () => {
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers/6${testSoldier.id}`,
+    });
+    expect(res.statusCode).toBe(404);
+  });
+
+  it('checks for status 500 when theres a server error', async () => {
+    const lookForSoldierSpy = vi.spyOn(database, 'lookForSoldier');
+    lookForSoldierSpy.mockImplementation(() => {
+      const err = new Error();
+      err.statusCode = 500;
+      throw err;
+    });
+    const res = await app.inject({
+      method: 'GET',
+      url: '/soldiers/:id',
+      body: testSoldier.id,
+    });
+    expect(res.json()).toEqual({ message: 'server error' });
+  });
+});
+
+describe('Get soldiers route', () => {
+  it('checks that route returns the relevant object', async () => {
+    await addNewSoldier(testSoldier);
+    const result = await app.inject({
+      method: 'GET',
+      url: `/soldiers?rank=${testSoldier.rank}`,
+    });
+    expect(result.body).toContain(testSoldier.name);
+  });
+
+  it('checks that when searched using id you still get an object back', async () => {
+    await addNewSoldier(testSoldier);
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers?name=${testSoldier.name}&id=${testSoldier.id}`,
+    });
+    expect(res).toBeTypeOf('object');
```

Comment:
keep in mind what is `res` and what is `res.json()`. is `res` ever not an object?

Reply:
got it. fixed